das/state.go

```diff
@@ -1,53 +1,223 @@
 package das
 
 import (
-	"sync"
-	"time"
+	"context"
 )
 
-// state collects information about the DASer process. Currently, there are
-// only two sampling routines: the main sampling routine which performs sampling
-// over current network headers, and the `catchUp` routine which performs sampling
-// over past headers from the last sampled checkpoint.
-type state struct {
-	sampleLk sync.RWMutex
-	sample   RoutineState // tracks information related to the main sampling routine
+// coordinatorState represents current state of sampling
+type coordinatorState struct {
+	rangeSize uint64
 
-	catchUpLk sync.RWMutex
-	catchUp   JobInfo // tracks information related to the `catchUp` routine
+	priority   []job                      // list of headers heights that will be sampled with higher priority
+	inProgress map[int]func() workerState // keeps track of running workers
+	failed     map[uint64]int             // stores heights of failed headers with amount of attempt as value
+
+	nextJobID   int
+	next        uint64 // all headers before next were sent to workers
+	networkHead uint64
+
+	catchUpDone   bool          // indicates if all headers are sampled
+	catchUpDoneCh chan struct{} // blocks until all headers are sampled
+}
+
+// initState initiates state for samplingCoordinator
+func initState(samplingRangeSize uint64) coordinatorState {
```

Comment:
Can we name this newState or newCoordinatorState or something? init sounds like an action is being taken